ntbs-service/Services/HospitalSearchService.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using ntbs_service.DataAccess;
+using ntbs_service.Models.ReferenceEntities;
+
+namespace ntbs_service.Services
+{
+    public interface IHospitalSearchService
+    {
+        Task<IList<Hospital>> OrderQueryableAsync(string searchKeyword);
+    }
+
+    class HospitalSearchService : IHospitalSearchService
+    {
+        private readonly IReferenceDataRepository _referenceDataRepository;
+
+        public HospitalSearchService(IReferenceDataRepository referenceDataRepository)
+        {
+            _referenceDataRepository = referenceDataRepository;
+        }
+
+        public async Task<IList<Hospital>> OrderQueryableAsync(string searchKeyword)
+        {
+            var searchKeywords = searchKeyword.Split(" ")
+                .Where(x => !x.IsNullOrEmpty())
+                .Select(s => s.ToLower()).ToList();
+            
+            var allHospitals = await _referenceDataRepository.GetAllHospitals();
```

Comment:
Here you're fetching the full list of hospitals from the database, and then searching it in memory in C#. This is not very efficient - it would be better to do the searching in the database itself.

Note that if you're doing the search in SQL then you don't need the `ToLower()` call as well, because SQL is case-insensitive

Reply:
Realistically, I'm not sure it's worth have a separate `HospitalSearchService` - it's doing so little that it's probably better to just call a new method in the `ReferenceDataRepository` directly from the calling class.